File: common/src/main/java/juuxel/adorn/platform/ItemGroupBridge.java
package juuxel.adorn.platform;

import juuxel.adorn.item.group.ItemGroupModifyContext;
import juuxel.adorn.util.InlineServices;
import juuxel.adorn.util.Services;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.resources.ResourceKey;

import java.util.function.Consumer;

@InlineServices
public interface ItemGroupBridge {
    CreativeModeTab.Builder builder();
    void addItems(ResourceKey<CreativeModeTab> group, Consumer<ItemGroupModifyContext> configurator);

    @InlineServices.Getter
    static ItemGroupBridge get() {
        return Services.load(ItemGroupBridge.class);
    }
}

File: common/src/main/kotlin/juuxel/adorn/platform/ItemGroupBridge.kt
package juuxel.adorn.platform

import juuxel.adorn.item.group.ItemGroupModifyContext
import juuxel.adorn.util.InlineServices
import juuxel.adorn.util.loadService
import net.minecraft.item.ItemGroup
import net.minecraft.registry.RegistryKey

interface ItemGroupBridge {
    fun builder(): ItemGroup.Builder
    fun addItems(group: RegistryKey<ItemGroup>, configurator: ItemGroupModifyContext.() -> Unit)

    @InlineServices
    companion object {
        private val instance: ItemGroupBridge by lazy { loadService() }
        fun get(): ItemGroupBridge = instance
    }
}
